tree/30_generic_methods/service_action.cf

```diff
@@ -39,7 +39,15 @@ bundle agent service_action(service_name, action)
 
       "action_command"            string => "${paths.path[service]} ${service_name} ${action}";
 
-    !systemctl_utility_present.!service_utility_present.init_d_directory_present::
+    !systemctl_utility_present.!service_utility_present.startsrc_utility_present::
+      "svc_action_cmd[restart]"   string => "/usr/bin/stopsrc -s ${service_name} && until lssrc -s ${service_name} | grep -q inoperative; do perl -e 'select(undef,undef,undef,.25)'; done; /usr/bin/startsrc -s ${service_name}";
```

Comment:
This is a mistake. We have to use full paths, because CFEngine (at least <= 3.6) checks if the first "string" in a command in a valid binary. The rest of the line doesn't stricly *have* to use full paths but it is good practice to avoid errors when refactoring.